upup/pkg/fi/cloudup/awstasks/securitygroup.go

```diff
@@ -268,34 +269,39 @@ type deleteSecurityGroupRule struct {
 var _ fi.Deletion = &deleteSecurityGroupRule{}
 
 func (d *deleteSecurityGroupRule) Delete(t fi.Target) error {
-	klog.V(2).Infof("deleting security group permission: %v", fi.DebugAsJsonString(d.permission))
+	if !featureflag.DeleteUnknownSGRules.Enabled() {
+		klog.Warningf("found unknown security group permission: %v", fi.DebugAsJsonString(d.permission))
+	} else {
 
-	awsTarget, ok := t.(*awsup.AWSAPITarget)
-	if !ok {
-		return fmt.Errorf("unexpected target type for deletion: %T", t)
-	}
+		klog.V(2).Infof("deleting security group permission: %v", fi.DebugAsJsonString(d.permission))
 
-	if d.egress {
-		request := &ec2.RevokeSecurityGroupEgressInput{
-			GroupId: d.groupID,
+		awsTarget, ok := t.(*awsup.AWSAPITarget)
+		if !ok {
+			return fmt.Errorf("unexpected target type for deletion: %T", t)
 		}
-		request.IpPermissions = []*ec2.IpPermission{d.permission}
 
-		klog.V(2).Infof("Calling EC2 RevokeSecurityGroupEgress")
-		_, err := awsTarget.Cloud.EC2().RevokeSecurityGroupEgress(request)
-		if err != nil {
-			return fmt.Errorf("error revoking SecurityGroupEgress: %v", err)
-		}
-	} else {
-		request := &ec2.RevokeSecurityGroupIngressInput{
-			GroupId: d.groupID,
-		}
-		request.IpPermissions = []*ec2.IpPermission{d.permission}
+		if d.egress {
+			request := &ec2.RevokeSecurityGroupEgressInput{
+				GroupId: d.groupID,
+			}
+			request.IpPermissions = []*ec2.IpPermission{d.permission}
 
-		klog.V(2).Infof("Calling EC2 RevokeSecurityGroupIngress")
-		_, err := awsTarget.Cloud.EC2().RevokeSecurityGroupIngress(request)
-		if err != nil {
-			return fmt.Errorf("error revoking SecurityGroupIngress: %v", err)
+			klog.V(2).Infof("Calling EC2 RevokeSecurityGroupEgress")
+			_, err := awsTarget.Cloud.EC2().RevokeSecurityGroupEgress(request)
+			if err != nil {
+				return fmt.Errorf("error revoking SecurityGroupEgress: %v", err)
```

Comment:
Consider wrapping the error:
```suggestion
				return fmt.Errorf("error revoking security group egress: %w", err)
```